lib/topological_inventory/ansible_tower/operations/worker.rb

```diff
@@ -4,29 +4,38 @@
 require "topological_inventory/ansible_tower/connection_manager"
 require "topological_inventory/ansible_tower/messaging_client"
 require "topological_inventory/providers/common/operations/health_check"
+require "topological_inventory/providers/common/operations/async_worker"
 
 module TopologicalInventory
   module AnsibleTower
     module Operations
       class Worker
         include Logging
 
+        ASYNC_MESSAGES = %w[Source.availability_check].freeze
+
         def run
-          TopologicalInventory::AnsibleTower::ConnectionManager.start_receptor_client
-          sleep 5
```

Comment:
Is there no longer a need for this `sleep 5` after `start_receptor_client` in `start_workers`?

Reply:
No I don't think so - there isn't a start-up delay with rdkafka like there was with the ruby-kafka library. 

With ruby-kafka it would do some message producing to let us know the client was ready to process messages etc. Since the new rdkafka library is a FFI C library the start-up time is basically nil. I can add a small timeout if we run into issues - but initially I just left it in there because we had a timeout before, it has been behaving fine locally in my testing.